rgbd/src/server.cpp

```diff
@@ -1,147 +1,34 @@
-            cv::Mat depth_image(depth_img_ptr->image.rows, depth_img_ptr->image.cols, CV_32FC1);
-            for(int x = 0; x < depth_image.cols; ++x)
-            {
-                for(int y = 0; y < depth_image.rows; ++y)
-                {
-                    depth_image.at<float>(y, x) = ((float)depth_img_ptr->image.at<unsigned short>(y, x)) / 1000; // (mm to m)
-                }
-            }
-
-            depth_img_ptr->image = depth_image;
-        }
-
-    } catch (cv_bridge::Exception& e) {
-        ROS_ERROR("Could not deserialize depth image: %s", e.what());
-        return;
-    }
-
-    rgbd::Image image(img_ptr->image,depth_img_ptr->image,cam_model_,rgb_image_msg->header.frame_id,rgb_image_msg->header.stamp.toSec());
-
-    rgbd_server.send(image);
+Server::Server()
+{
 }
 
 // ----------------------------------------------------------------------------------------
 
-void camInfoCallback(const sensor_msgs::CameraInfoConstPtr& cam_info_msg) {
-    if (!cam_model_.initialized()) {
-        cam_model_.fromCameraInfo(cam_info_msg);
-    }
+Server::~Server()
+{
 }
 
 // ----------------------------------------------------------------------------------------
 
-int main(int argc, char **argv) {
-    ros::init(argc, argv, "rgbd_transport_server");
-
-    ros::NodeHandle nh;
-    ros::NodeHandle nh_private("~");
-
-    double max_fps = 30;
-    nh_private.getParam("max_fps", max_fps);
-
-    // ----- READ RGB STORAGE TYPE
-
-    rgbd::RGBStorageType rgb_type;
-
-    std::string rgb_type_str = "lossless";
-    nh_private.getParam("rgb_storage", rgb_type_str);
-    if (rgb_type_str == "none")
-        rgb_type = rgbd::RGB_STORAGE_NONE;
-    else if (rgb_type_str == "lossless")
-        rgb_type = rgbd::RGB_STORAGE_LOSSLESS;
-    else if (rgb_type_str == "jpg")
-        rgb_type = rgbd::RGB_STORAGE_JPG;
-    else
-    {
-        ROS_ERROR("Unknown 'rgb_storage' type: should be 'none', 'lossless', or 'jpg'.");
-        return 1;
-    }
-
-    // ----- READ DEPTH STORAGE TYPE
-
-    rgbd::DepthStorageType depth_type;
-
-    std::string depth_type_str = "lossless";
-    nh_private.getParam("depth_storage", depth_type_str);
-    if (depth_type_str == "none")
-        depth_type = rgbd::DEPTH_STORAGE_NONE;
-    else if (depth_type_str == "lossless")
-        depth_type = rgbd::DEPTH_STORAGE_LOSSLESS;
-    else if (depth_type_str == "png")
-        depth_type = rgbd::DEPTH_STORAGE_PNG;
-    else
-    {
-        ROS_ERROR("Unknown 'depth_storage' type: should be 'none', 'lossless', or 'png'.");
-        return 1;
-    }
-
-    // ------------------------------
-
-    rgbd_server.initialize(ros::names::resolve("rgbd"), rgb_type, depth_type);
-
-    ros::Subscriber sub_cam_info = nh.subscribe("cam_info", 1, &camInfoCallback);
+void Server::initialize(const std::string& name, RGBStorageType rgb_type, DepthStorageType depth_type)
+{
+    server_rgbd_.initialize(name, rgb_type, depth_type);
+    server_shm_.initialize(name);
```

Comment:
Can these things fail?

Reply:
Not really, also not returning a bool. Might change with the reconnect, but right now not.